framework/branch-key-store.md

```diff
@@ -74,29 +88,72 @@ A list of AWS KMS [grant tokens](https://docs.aws.amazon.com/kms/latest/develope
 
 The DynamoDb Client used to put and get keys from the backing DDB table.
 
-If not provided, one will be created in the region of the supplied KMS Key ARN.
+If the AWS KMS Configuration is KMS Key ARN or KMS MRKey ARN,
+and no DynamoDb Client is provided,
+a new DynamoDb Client MUST be created
+with the region of the supplied KMS ARN.
+
+If the AWS KMS Configuration is Discovery,
+and no DynamoDb Client is provided,
+a new DynamoDb Client MUST be created
+with the default configuration.
+
+If the AWS KMS Configuration is MRDiscovery,
+and no DynamoDb Client is provided,
+a new DynamoDb Client MUST be created
+with the region configured in the MRDiscovery.
 
 ### KMS Client
 
 The KMS Client used when wrapping and unwrapping keys.
 
-If not provided, one will be created in the region of the supplied KMS Key ARN.
+If the AWS KMS Configuration is KMS Key ARN or KMS MRKey ARN,
+and no KMS Client is provided,
+a new KMS Client MUST be created
+with the region of the supplied KMS ARN.
+
+If the AWS KMS Configuration is Discovery,
+and no KMS Client is provided,
+a new KMS Client MUST be created
+with the default configuration.
+
+If the AWS KMS Configuration is MRDiscovery,
+and no KMS Client is provided,
+a new KMS Client MUST be created
+with the region configured in the MRDiscovery.
 
-On initialization the KeyStore MUST append a user agent string to the AWS KMS SDK Client with the
-value `aws-kms-hierarchy`.
+On initialization the KeyStore SHOULD
+append a user agent string to the AWS KMS SDK Client with
+the value `aws-kms-hierarchy`.
 
 ### Table Name
 
 The table name of the DynamoDb table that backs this Keystore.
 
 ### AWS KMS Configuration
 
-A valid [AWS KMS Key ARN](./aws-kms/aws-kms-key-arn.md#a-valid-aws-kms-arn)
-that wraps and unwraps keys stored in Amazon DynamoDB.
+This configures the Keystore's KMS Key ARN restrictions,
+which determines which KMS Key(s) is used
+to wrap and unwrap the keys stored in Amazon DynamoDB.
+There are four (4) options:
 
-The KMS Configuration MUST distinguish between Single Region Key ARN compatibility and Multi Region Key (MRK) ARN compatibility.
+- Discovery
+- MRDiscovery
+- Single Region Key Compatibility, denoted as `KMS Key ARN`
+- Multi Region Key Compatibility, denoted as `KMS MRKey ARN`
 
-Both compatibility modes are allowed with both MRK ARN's and Single Region ARNs.
+`KMS Key ARN` and `KMS MRKey ARN` MUST take an additional argument
+that is a KMS ARN.
+This ARN MUST NOT be an Alias.
+This ARN MUST be a valid
+[AWS KMS Key ARN](./aws-kms/aws-kms-key-arn.md#a-valid-aws-kms-arn).
+
+Both `KMS Key ARN` and `KMS MRKey ARN` accept MRK or regular Single Region KMS ARNs.
```

Comment:
It might be worth expanding for clarity. I would have assumed that `KMS Key ARN` configurations MUST NOT accept MRK ARNs, 
and that `KMS MRK ARN` configurations MUST NOT accept single Region ARNs. 
But this appears to not be the case?

Reply:
```suggestion
Both `KMS Key ARN` and `KMS MRKey ARN` accept MRK or regular Single Region KMS ARNs.

To be clear, an KMS ARN for a Multi-Region Key MAY be provided to the `KMS Key ARN` configuration,
and a KMS ARN for non Multi-Region Key MAY be provided to the `KMS MRKey ARN` configuration.
```